src/view/SmartQuestionList.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import questionModel from "../model/QuestionModel"
+import QuestionList from './QuestionList';
+import questionsListPresenter from '../presenter/questionsListPresenter';
+
+const mapModelStateToComponentState = modelState => ({
+    questions: modelState.questions
+});
+
+export default class SmartQuestionsList extends Component {
+    constructor() {
+        super();
+        this.state = {
+            ...mapModelStateToComponentState(questionModel.state),
+            titleFilter: '',
+            tagsFilter: '',
+            results: [],
+        };
+        this.listener = modelState => this.setState(mapModelStateToComponentState(modelState));
+        questionModel.addListener("change", this.listener);
+    }
+
+
+    componentWillUnmount() {
+        questionModel.removeListener("change", this.listener);
+    }
+
+    getQuestions() {
+        const { titleFilter, tagsFilter, results, questions } = this.state;
+        return (titleFilter || tagsFilter) ? results : questions;
+    }
+
+    render() {
+        return (
+            <QuestionList
+                onCreateQuestion={questionsListPresenter.onCreateQuestion}
+                questions={this.getQuestions()}
+                applyTitleFilter={event => {
+                    const titleFilter = event.target.value;
```

Comment:
This should be placed either in the presenter, or even better, inside the model. The Smart component should just act as a bridge between the model, presenter and dumb component (it should not hold it's own independent state nor have any kind of logic).